### PR TITLE
Bug CORE-498 | Unable to update channels

### PR DESCRIPTION
**Pull request by LucasStarlingdePaulaSalles**
# Description

Kind: Bug

Section: Diff

Summary:
There were two problems causing the misbehavior:
- The diff for metadata was only processing Create and Delete changes on `Annotations`
- The diff for metadata was flagging `Annotations` changes as reaction differences, since we no longer need the annotations on our Kafka topics, this is no longer necessary.

## Changelog
- fixes:
    - Changed the Annotations metadata diff to work according to new brokers configuration process.
    - Copied `SelectedBroker` from old channel to new on Update operation
- misc:
    - changed Skaffod deployment to get secret values from a file other than the default Values.

### CODE REVIEW

File: pkg/api/handlers/handler.go [hunk 1]
```diff
@@ -58,6 +59,8 @@ func (handler *Handler) addChangeReactor(op ...diff.ChangeReaction) {
 }
 
 func (handler *Handler) applyChangesInDiff(changes diff.Changelog) error {
+	logger.Debug("trying to aplly changes in diff",
```
Comment (franciscobonand) [inline pkg/api/handlers/handler.go:62]: "apply"*

Comment (franciscobonand) [inline pkg/api/handlers/handler.go]: "apply"*

File: cmd/insprd/memory/tree/channel.go [hunk 2]
```diff
@@ -209,6 +209,9 @@ func (chh *ChannelMemoryManager) Update(scope string, ch *meta.Channel) error {
 	ch.ConnectedAliases = oldCh.ConnectedAliases
 	ch.Meta.UUID = oldCh.Meta.UUID
 
+	broker := oldCh.Spec.SelectedBroker
+	ch.Spec.SelectedBroker = broker
```
Comment (luiz-couto) [inline cmd/insprd/memory/tree/channel.go]: Why didn't you do ch.Spec.SelectedBroker = oldCh.Spec.SelectedBroker?

File: cmd/insprd/memory/tree/manager.go [hunk 3]
```diff
@@ -16,6 +16,7 @@ var logger *zap.Logger
 // have been initialized
 func init() {
 	logger, _ = zap.NewProduction(zap.Fields(zap.String("section", "memory-tree")))
+	// logger, _ = zap.NewDevelopment(zap.Fields(zap.String("section", "memory-tree")))
```
Comment (luiz-couto) [inline cmd/insprd/memory/tree/manager.go:19]: Can this comment be removed?

**Review by franciscobonand — APPROVED** [review]
Looks good. There's a misspell that I pointed out, and I'd like you to explain more in detail the diff.go change for me please.  

Also, the test/coverage is dropping significantly, it's a good idea to review the tests and update them if needed.

**Review by nicholaspcr — APPROVED** [review]

**Review by luiz-couto — APPROVED** [review]

**Review by lucasABLima — APPROVED** [review]

**Pull request merged**